[EA Forum only] email inactive users a feedback survey

s-cheng opened the pull request:
This PR sets up cron job that emails inactive users a feedback survey. (Currently this is users who haven't read a post in the past 3 months.) Each user will only ever receive this email once. More details in [this doc](https://docs.google.com/document/d/1i3hdI48X8w6gmsU_fM3LMfQLD_qInJtsxtx9x7QImJU/edit).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206826240903636) by [Unito](https://www.unito.io)

s-cheng commented on packages/lesswrong/server/inactiveUserSurveyCron.tsx:
```diff
@@ -0,0 +1,57 @@
+  
+  const logger = loggerConstructor(`cron-sendInactiveUserSurveyEmails`)
+  
+  // Get up to 30 inactive users to email
+  const users = await new UsersRepo().getInactiveUsersToEmail(30)
```
It looks like each day we get about 8 users that would qualify for this email. I think many of those are uncaught spam. I'm not sure why they are uncaught since I thought our forum facilitators looked at all new users who write something in their profile.

In any case, I would prefer not to email spammers but I think it's better to do that than to miss real users. I'm limiting it to 30 per day for now in case there are bugs, and there is no urgency to sending this email. I'll increase this limit later if there are no issues.

s-cheng commented on packages/lesswrong/server/userJobAdCron.tsx:
```diff
@@ -62,12 +65,14 @@ const sendJobAdReminderEmails = async () => {
+      void sendJobAdReminderEmails();
+    }
+  });
+}
```
Forgot to forum-gate this, oops. Didn't cause any issues though since no other sites have job ads.

jpaddison3 commented on packages/lesswrong/server/repos/UsersRepo.ts:
```diff
@@ -582,6 +582,51 @@ class UsersRepo extends AbstractRepo<"Users"> {
+        AND u."deleteContent" IS NOT TRUE
+        AND u."sunshineFlagged" IS NOT TRUE
+        AND (
+          u."reviewedByUserId" IS NOT NULL
```
You could also filter out snoozed users, but that may end up missing out on real users.

jpaddison3 commented on packages/lesswrong/server/repos/UsersRepo.ts:
```diff
@@ -582,6 +582,51 @@ class UsersRepo extends AbstractRepo<"Users"> {
+        AND u."isAdmin" IS NOT TRUE
+        AND u.deleted IS NOT TRUE
+        AND u."deleteContent" IS NOT TRUE
+        AND u."sunshineFlagged" IS NOT TRUE
```
Filter out banned users. Would be quite awkward.

jpaddison3 commented on packages/lesswrong/server/emailComponents/EmailInactiveUserSurvey.tsx:
```diff
@@ -0,0 +1,57 @@
+}) => {
+
+  return (
+    <div className={classes.root}>
```
Did you test the preview of this email?

jpaddison3 replied:
(I know you're gonna edit it, and probably testing it after that is good too, but still, good to check the visuals. I would also include a way to send an actual test email to an admin account.)

s-cheng replied:
Yeah I tested it, and here is an updated version with the new copy:

![Screen Shot 2024-03-14 at 3 35 23 PM](https://github.com/ForumMagnum/ForumMagnum/assets/9057804/d361aeba-22c6-4809-894b-aa1e1ab481e5)

I'm not going to bother adding a way to send a test email to an admin account - the contents of this email are static, that functionality seems more useful to add for dynamic emails.

jpaddison3 reviewed (approved):
A few comments, but basically this looks good, and thanks!

s-cheng commented:
@jpaddison3 I've changed the threshold to be 6 months (may decrease it in the future), and updated the query and copy. I'll ask Heidi to copy-edit my doc so I won't be able to deploy this for a while. It might be helpful for me if you double check the current code before then, but if you don't have time then no worries.
